fix support for elasticsearch v7 + add types

robyte-ctrl opened the pull request:
opened the PR in here by mistake (meant to merge it in my fork's `master` branch) so feel free to close it

but this PR does fix https://github.com/pinojs/pino-elasticsearch/issues/166 and it also adds types so it's easier to use this library in a typescript project

mcollina reviewed (commented):
Good work!

This is a semver-major change, right?

robyte-ctrl commented:
> Good work!
> 
> This is a semver-major change, right?

@mcollina shouldn't be, although I have renamed some of the options so that they follow the camelCase naming notation, the old names still work and I've added some tests to make sure that they do

if you want to remove the old names completely then that would be a semver-major change

mcollina commented:
I see you have removed --bulk-size from the CLI options, hence my guess.

robyte-ctrl commented on lib.d.ts:
```diff
@@ -0,0 +1,43 @@
+import type { Transform } from 'stream'
```
done

though it's first time I tried tests for types so I'm not sure how good they are
